tests/unit/runner/test_runner.py

```diff
@@ -135,13 +86,10 @@ async def test_runner_app_state_pending():
 )
 async def test_runner_app_state_starting(nodes_states):
     """Test app state reporting while Runner is starting its services."""
-    dapp_node_count = 2
-    desired_app_state = ServiceState.running
 
-    assert (
-        Runner._get_app_state_from_nodes(dapp_node_count, desired_app_state, nodes_states)
-        == ServiceState.starting
-    )
+    with mock.patch("dapp_runner.runner.Runner.dapp_state", nodes_states):
```

Comment:
Consider using `mocker` fixture from pytest, as it gives better UX with mocking.

Reply:
in what way does it give better UX? asking just out of curiosity?